### PR TITLE
Implement Encode and Decode for empty arrays

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

**Review by thiolliere — APPROVED** [review]

File: src/codec.rs [hunk 1]
```diff
@@ -665,7 +667,7 @@ macro_rules! impl_array {
 }
 
 impl_array!(
-	1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
+	0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
```
Comment (ordian) [inline src/codec.rs:670]: it should not require `T: Encode + Decode` though for empty arrays, right?

Comment (bkchr) [reply src/codec.rs:670]: While that is conceptually right, why would you want to write such code? Than you directly could use `()`.

Comment (ordian) [reply src/codec.rs:670]: Good point, I don't know what @Robbepop use-case is.

Comment (Robbepop) [reply src/codec.rs:670]: Before we merge this: I was able to resolve the problem without having support for SCALE encoded empty arrays. If you still want to merge this, go ahead.

Comment (bkchr) [reply src/codec.rs:670]: Fine to merge anyway :)

Comment (Robbepop) [reply src/codec.rs:670]: Yes, it is a nice generalization imo.

**Pull request merged**